libnetdata/popen/popen.c

```diff
@@ -116,7 +117,7 @@ static inline FILE *custom_popene(const char *command, volatile pid_t *pidptr, c
     if (posix_spawn_file_actions_destroy(&fa))
         error("posix_spawn_file_actions_destroy");
 error_after_pipe:
-    close(pipefd[PIPE_READ]);
+    fclose(fp);
```

Comment:
This will not work because if `fdopen()` fails you need to `close()` the pipe and not `fclose()` it.

You must change it to:
```
    if (fp)
        fclose(fp);
    else
        close(pipefd[PIPE_READ]);
```
```

Reply:
Thank you @mfundul , I really missed this.